tests/roles/telemetry_adoption/tasks/main.yaml

```diff
@@ -0,0 +1,80 @@
+- name: deploy cluster-observability-operator
+  ansible.builtin.shell: |
+    {{ shell_header }}
+    {{ oc_header }}
+    oc create -f - <<EOF
+    apiVersion: operators.coreos.com/v1alpha1
+    kind: Subscription
+    metadata:
+      name: cluster-observability-operator
+      namespace: openshift-operators
+    spec:
+      channel: development
+      installPlanApproval: Automatic
+      name: cluster-observability-operator
+      source: redhat-operators
+      sourceNamespace: openshift-marketplace
+    EOF
+    '
+
+- name: deploy metric storage
+  ansible.builtin.shell: |
+    {{ shell_header }}
+    {{ oc_header }}
+    oc patch openstackcontrolplane openstack --type=merge --patch '
+    spec:
+      telemetry:
+        enabled: true
+        template:
+          metricStorage:
+            enabled: true
+            monitoringStack:
+              alertingEnabled: true
+              scrapeInterval: 30s
+              storage:
+                strategy: persistent
+                retention: 24h
+                persistent:
+                  pvcStorageRequest: 20G
+    '
+
+- name: wait for alertmanager metric storage to start up
+  ansible.builtin.shell: |
+    {{ shell_header }}
+    {{ oc_header }}
+    oc get pod --selector=alertmanager=metric-storage -o jsonpath='{.items[0].status.phase}{"\n"}' | grep Running
+  register: alertmanager_running_result
+  until: alertmanager_running_result is success
+  retries: 60
+  delay: 2
+
+- name: wait for prometheus metric storage to start up
+  ansible.builtin.shell: |
+    {{ shell_header }}
+    {{ oc_header }}
+    oc get pod --selector=prometheus=metric-storage -o jsonpath='{.items[0].status.phase}{"\n"}' | grep Running
+  register: prometheus_running_result
+  until: prometheus_running_result is success
+  retries: 60
+  delay: 2
+
+- name: deploy podified Ceilometer
+  ansible.builtin.shell: |
+    {{ shell_header }}
+    {{ oc_header }}
+    oc patch openstackcontrolplane openstack --type=merge --patch '
+    spec:
+      telemetry:
+        enabled: true
+        secret: osp-secret
+    '
+
+- name: wait for Ceilometer to start up
+  ansible.builtin.shell: |
+    {{ shell_header }}
+    {{ oc_header }}
+    oc get pod --selector=service=ceilometer -o jsonpath='{.items[0].status.phase}{"\n"}' | grep Running
+  register: ceilometer_running_result
+  until: ceilometer_running_result is success
+  retries: 60
+  delay: 2
```

Comment:
I would say that we should have here a new patch that adds the logging section with enabled: false, but with the fields like this:

logging:
        enabled: false
        network: internalapi
        ipaddr: 172.17.0.80
        port: 10514
        cloNamespace: openshift-logging

this would give the user a good reference for the future if he wants to enable logging.

Reply:
If we add logging with enabled: false what should be checked in the next task ? For other tasks we verify whether the pods of those service are available or not.

Reply:
I dont think we have to check anything, as no pod will be running after adding logging in enabled: false. We just leave this on the openstackcontrolplane CR for the user to manually enable logging in the future.